msal_extensions/libsecret.py

```diff
@@ -111,9 +127,12 @@ def trial_run():
         assert agent.load() == payload  # This line is probably not reachable
         agent.clear()
     except (gi.repository.GLib.Error, AssertionError):
-        message = (
-            "libsecret did not perform properly. Please refer to "
-            "https://github.com/AzureAD/microsoft-authentication-extensions-for-python/wiki/Encryption-on-Linux")  # pylint: disable=line-too-long
+        message = """libsecret did not perform properly.
+* If you encountered error "Remote error from secret service:
+  org.freedesktop.DBus.Error.ServiceUnknown",
+  you may need to install gnome-keyring package.
+* Headless mode (such as in an ssh session) is not supported.
```

Comment:
Just for my own understanding: is this because GNOME tries to display UI?

Reply:
In fact, the UI display seems only needed/visible at the first interaction during a desktop session, after that it works silently. My understanding is the secret service is somehow only activated in an X session, otherwise you will see this error:
> gi.repository.GLib.GError: g-io-error-quark: Cannot autolaunch D-Bus without X11 $DISPLAY (0)